CONTRIBUTING.md

```diff
@@ -0,0 +1,169 @@
+# Contributing to Puppet modules
+
+So you want to contribute to a Puppet module: Great! Below are some instructions
+to get you started while setting expectations around code quality, as well as a
+few tips for making the process as easy as possible. 
+
+## Community Slack Channels
+
+Join the `#bolt` channel in the [Puppet community
+Slack](https://slack.puppet.com/) where Bolt developers and community members
+who use and contribute to Bolt discuss the tool.
+
+## Getting started
+
+- Fork the module repository on GitHub and clone to your workspace
+- Make your changes!
+
+## Commit checklist
+
+### The basics
+
+- My commit is a single logical unit of work
+- I have checked for unnecessary whitespace with `git diff --check`
+- My commit does not include commented out code or unneeded files
+
+### The content
+
+- My commit includes tests for the bug I fixed or feature I added
+- My commit includes appropriate documentation changes if it is introducing a
+  new feature or changing existing functionality
+- My code passes existing test suites
+
+### The commit message
+
+- The first line of my commit message includes:
+  - An issue number (if applicable), e.g. "(GH-xxxx) This is the first line"
+  - A short description (50 characters is the soft limit, excluding ticket
+    number(s))
+- The body of my commit message:
+  - Is meaningful
+  - Uses the imperative, present tense: "change", not "changed" or "changes"
+  - Includes motivation for the change, and contrasts its implementation with
+    the previous behavior
+
+## Testing
+
+### Getting started
+
+Our Puppet modules provide [`Gemfile`](./Gemfile)s, which can tell a Ruby
+package manager such as [bundler](http://bundler.io/) what Ruby packages, or
+Gems, are required to build, develop, and test this software.
+
+Please make sure you have [bundler
+installed](http://bundler.io/#getting-started) on your system, and then use it
+to install all dependencies needed for this project in the project root by
+running
+
+```shell
+$ bundle install --path .bundle/gems
+```
+
+> **NOTE:** some systems may require you to run this command with sudo.
+
+If you already have those gems installed, make sure they are up-to-date:
```

Comment:
```suggestion
If you already have those gems installed, make sure they are up to date:
```
Nerdy grammar rule: If a compound adjective doesn't come immediately before the noun, it usually doesn't require hyphenation.